MapboxNavigation/InstructionPresenter.swift

```diff
@@ -186,7 +186,11 @@ class InstructionPresenter {
         
         let view = ExitView(pointSize: dataSource.font.pointSize, side: side, text: text)
         let attachment = ExitAttachment()
-        guard let cacheKey = component.cacheKey() else { return nil }
+        guard var cacheKey = component.cacheKey() else { return nil }
+        
+        if let foregroundColor = view.foregroundColor {
+            cacheKey = "\(cacheKey)-\(String(foregroundColor.cgColor.hashValue))"
```

Comment:
So I was taking a look at this earlier today... We _could_ key off of the active style, but it would be a PITA to pass the reference to the `StyleManager` down to the `InstructionPresenter` with the current architecture, totally agree with your assessment.

So with that solution being a no-go, what I would probably suggest here is to implement a custom `hashValue` for `ExitView`. NSHipster [defines](http://nshipster.com/equality/) a good hashing mechanism as "one that XOR's over all critical properties", which I would define in exitView like this: 

```swift
override var hashValue: Int {
    get {
      return [foregroundColor, backgroundColor, side, exitText, pointSize].map({ $0.hashValue }).reduce(0, ^)
    }
}
```

With this logic, we'll cover _all_ our bases, not just `foregroundColor`.

Also: Hash values are [not guaranteed](https://developer.apple.com/documentation/swift/hashable) to be consistent across different runs of your app, so we definitely shouldn't be persisting this cache across different app runs. It looks like we [don't](https://github.com/mapbox/mapbox-navigation-ios/pull/1398/files#diff-af48d70309ee195e34cc5e6cc6b27caaL195) disk cache, but I wanted to call that out anyways, as it's a really nasty pitfall of `Hashable`.